src/components/Proposals/ProposalPage/ApprovedTransactions/OptionDescription.jsx

```diff
@@ -1,13 +1,19 @@
 import TokenAmountDisplay from "@/components/shared/TokenAmountDisplay";
 import { shortAddress } from "@/lib/utils";
+import Tenant from "@/lib/tenant/tenant";
 
 export default function OptionDescription({ description, value, target }) {
+  const { token } = Tenant.current();
+
   return (
     <span>
-      {/* TODO: Warning – this dangerously assumes all tokens are OP. For now, ok, but won't always be the case */}
       {"//"} {description} requesting{" "}
-      <TokenAmountDisplay amount={value} decimals={18} currency="OP" /> transfer
-      to {shortAddress(target)}
+      <TokenAmountDisplay
```

Comment:
Nice! Another option is to move all of the tenant token stuff into the TokenAmountDisplay component.

Reply:
Went for the refactor in this commit: https://github.com/voteagora/agora-next/commit/4c88c9e5f2a6146fbdbdb7a6b308a8e20be7c3ca lmk what do you think :)